modules/core/24-host/parse.go

```diff
@@ -32,6 +32,40 @@ func ParseIdentifier(identifier, prefix string) (uint64, error) {
 	return sequence, nil
 }
 
+// MustParseClientStatePath returns the client ID from a client state path. It panics
+// if the provided path is invalid or if the clientID is empty.
+func MustParseClientStatePath(path string) string {
+	clientID, err := parseClientStatePath(path)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return clientID
+}
+
+// parseClientStatePath returns the client ID from a client state path. It returns
+// an error if the provided path is invalid.
+func parseClientStatePath(path string) (string, error) {
```

Comment:
I debated whether to make this public or not. I opted for only making `MustParseClientStatePath` public. My reasoning is that when receiving an error from this function, it is difficult to tell whether the error was because this is not a client state path or because the key path itself is corrupted (clientID is malformed). I think it is safer to expect the caller to ensure that it only calls this function on paths it expects to be a client state path, thus only exposing `MustParseClientStatePath` makes sense as any error implies state corruption